components/new-member-list/index.js

```diff
@@ -4,26 +4,40 @@ import React from 'react';
 import classNames from 'components/new-member-list/new-member-list.module.scss';
 import { membersContext } from 'store/members/members-context';
 
-const NewMemberList = () => {
+const NewMemberList = ({ searchTerm }) => {
   const {
     state: { newMembersArr }
   } = membersContext();
   if (newMembersArr) {
     return (
       <div className={classNames.container}>
-        {newMembersArr.map((ele) => (
-          <React.Fragment key={ele['id']}>
-            {(ele.first_name || ele.username) && <NewMemberListItem newMemberDetails={ele} />}
-          </React.Fragment>
-        ))}
+        {newMembersArr.length > 0 &&
+          newMembersArr
+            .filter((ele) => {
+              if (searchTerm === '') {
```

Comment:
The same code is repeated twice and most of the if condition are returning the ele itself , could you look into having one util file and look for a way to make the function more concise